static/app/views/organizationIntegrations/integrationExternalTeamMappings.tsx

```diff
@@ -111,10 +127,13 @@ class IntegrationExternalTeamMappings extends AsyncComponent<Props, State> {
 
   handleResults = (results, mappingKey?: string) => {
     if (mappingKey) {
+      const {queryResults} = this.state;
       this.setState({
         queryResults: {
-          ...this.state.queryResults,
-          [mappingKey]: results,
+          ...queryResults,
+          // This line will keep a running collection of the queries made on this mappingKey
+          // That way, no matter what they type after making a selection, we don't lose the result
+          [mappingKey]: uniqBy([...results, ...(queryResults[mappingKey] ?? [])], 'id'),
```

Comment:
This merge is complicated enough that I'd suggest extracting it to a function, even if that function is still a one-liner.

Reply:
Created a separate function with a doc string 👍